src/trace/context.ts

```diff
@@ -238,6 +245,162 @@ export function readTraceFromSQSEvent(event: SQSEvent): TraceContext | undefined
   return;
 }
 
+export function readTraceFromSNSSQSEvent(event: SQSEvent): TraceContext | undefined {
+  if (event.Records && event.Records[0] && event.Records[0].body) {
+    try {
+      const parsedBody = JSON.parse(event.Records[0].body) as SNSMessage;
+      if (
+        parsedBody.MessageAttributes &&
+        parsedBody.MessageAttributes._datadog &&
+        parsedBody.MessageAttributes._datadog.Value
+      ) {
+        const traceData = JSON.parse(parsedBody.MessageAttributes._datadog.Value);
+        const traceID = traceData[traceIDHeader];
+        if (typeof traceID !== "string") {
+          return;
+        }
+        const parentID = traceData[parentIDHeader];
+        if (typeof parentID !== "string") {
+          return;
+        }
+        const sampledHeader = traceData[samplingPriorityHeader];
+        if (typeof sampledHeader !== "string") {
+          return;
+        }
```

Comment:
Nit: can we do a single type check for all three variables instead of three separate conditions?